intel_extension_for_transformers/llm/runtime/graph/models/chatglm/chatglm.cpp

```diff
@@ -106,128 +112,139 @@ static bool chatglm_model_eval_internal(model_context& lctx, const model_token*
   int rope_dim = head_size / 2;
   for (int il = 0; il < n_layer; ++il) {
     struct ne_tensor* cur;
-    struct ne_tensor* alpha = ne_new_f32(ctx0, std::sqrt(2.f * n_layer));
+    struct ne_tensor *alpha = ne_new_f32(ctx0, std::sqrt(2.f * n_layer));
 
     lctx.use_buf(ctx0, 0);
-
-    // LayerNorm::forward
+    
     cur = ne_norm(ctx0, inpL);
+    
+    ne_set_name(cur, "cur");
     cur = ne_mul(ctx0, cur, model.layers[il].norm[0]);
     cur = ne_add(ctx0, cur, model.layers[il].norm[1]);
-
+    
     struct ne_tensor* attn_input = cur;
     // GLM2SelfAttention
     {
       // Linear::forward compute QKV
       cur = ne_mul_mat(ctx0, model.layers[il].attn[0], cur);
       cur = ne_add(ctx0, cur, model.layers[il].attn[1]);
+      //cur = ne_add(ctx0, ne_repeat(ctx0, model.layers[il].attn[1], cur), cur);  // ChatGLM-1 [qlen, 3 * hidden] , ChatGLM-2 [qlen, hidden + 2 * kv_hidden]
 
-      ne_tensor* query_layer = ne_view_3d(ctx0, cur, head_size, n_head, N, 3 * head_size * ne_element_size(cur),
-                                          cur->nb[1], 0);  // [qlen, 3 * hidden]
+      // ChatGLM-1
+      ne_tensor *query_layer = ne_view_3d(ctx0, cur, head_size, n_head, N, 3 * head_size * ne_element_size(cur), cur->nb[1], 0); // [qlen, 3 * hidden]
 
       ne_set_name(query_layer, "query_layer");
+      //query_layer = ne_rope_inplace(ctx0, query_layer, n_past, rope_dim, 0);  // ChatGLM-2
```

Comment:
I will prefer default parameters as we talked before
```
NE_API struct ne_tensor* ne_rope_inplace(struct ne_context* ctx, struct ne_tensor* a, int n_past, int n_dims, int mode, int n_ctx = 0);
```

or you can define a new function like
```
NE_API struct ne_tensor* ne_rope_inplace_real(struct ne_context* ctx, struct ne_tensor* a, int n_past, int n_dims, int mode, int n_ctx);
```

Reply:
revert changes in gptj and llama with whatever option you choose

Reply:
I've rebuilt fully extended tests on the branch of this PR. 
https://inteltf-jenk.sh.intel.com/blue/organizations/jenkins/ITREX-cpp-graph-extension/detail/ITREX-cpp-graph-extension/84/pipeline

As we discussed, adding the dfault parameters for this c file may introduce more unnecessary code because native C does not support default parameters for functions. 

Can we keep the same as ggml rope kernel, which means no change?